public/request/set-claim/update-claim.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+require_once __DIR__ . '/../../../vendor/autoload.php';
+require_once __DIR__ . '/../../../lib/bootstrap.php';
+
+use RA\ArticleType;
+use RA\Permissions;
+
+$claimOwner = requestInputPost('o', null);
+$claimID = requestInputPost('i', null, 'integer');
+$gameID = requestInputPost('g', null, 'integer');
+$claimType = requestInputPost('c', null, 'integer'); // 0 - Primary, 1 - Collaboration
+$setType = requestInputPost('s', null, 'integer'); // 0 - New set, 1 - Revision
+$status = requestInputPost('t', null, 'integer'); // 0 - Active, 1 - Complete, 2 - Dropped
+$special = requestInputPost('e', null, 'integer'); // Special flag
+$claimDate = requestInputPost('d', null); // Claim date
+$doneDate = requestInputPost('f', null); // Done date
+
+if (authenticateFromCookie($user, $permissions, $userDetails, Permissions::Admin)) {
+    if (updateClaim($claimID, $claimType, $setType, $status, $special, $claimDate, $doneDate)) {
+        $comment = "$user updated $claimOwner's claim. ";
+        $comment .= "Claim Type: " . ($claimType == 0 ? "Primary. " : "Collaboration. ");
+        $comment .= "Set Type: " . ($setType == 0 ? "New. " : "Revision. ");
+        switch ($status) {
+            case 0:
+                $comment .= "Claim Status: Active. ";
+                break;
+            case 1:
+                $comment .= "Claim Status: Complete. ";
+                break;
+            case 2:
+                $comment .= "Claim Status: Dropped. ";
+                break;
+            default:
+                $comment .= "Claim Status: Active. ";
+                break;
+        }
+        $comment .= "Special: " . $special . ". ";
+        $comment .= "Claim Date: " . $claimDate . ". ";
+        $comment .= "Finished date: " . $doneDate . ".";
```

Comment:
It would be preferable to only report the changed fields. I only changed one field here, can you guess which?
> Jamiras updated Jamiras's claim. Claim Type: Primary. Set Type: Revision. Claim Status: Active. Special: 0. Claim Date: 2022-06-07 16:13:04. Finished date: 2022-09-07 16:13:04.

Reply:
Updated to only report the updated fields.